pydantic/_internal/_generate_schema.py

```diff
@@ -1181,7 +1181,7 @@ def apply_single_annotation(  # noqa C901
     if metadata_schema is not None:
         return metadata_schema
 
-    metadata_get_schema = getattr(metadata, '__get_pydantic_core_schema__', None)
```

Comment:
If we keep the `__pydantic_core_schema__` thing maybe we should keep the `__get_pydantic_core_schema__` thing? I would think we either support both or neither.. Could be convinced otherwise if there's benefits for practicality

Reply:
Is `__pydantic_core_schema__` another function we support?

Reply:
I see your trick below of having one handle the case where it is the type directly, and another where it is modifying an existing schema. Given that, I think we should probably drop the `__pydantic_core_schema__` check above, and just _only_ use `__modify_pydantic_core_schema__`. Unless there is some practical reason I'm missing.